nmostesting/mocks/Node.py

```diff
@@ -54,6 +66,246 @@ def get_sender(self, stream_type="video"):
+            response_data['transport_params'] = activations['staged']['transport_params']
+
+        # Set up default transport parameters to fill in auto or missing values
+        default_params = {'multicast_ip': None, 'destination_port': 5004, 'source_ip': get_default_ip(),
+                          'interface_ip': get_default_ip(), 'rtp_enabled': True, 'source_port': 5004}
+
+        # Update linked sender or receiver if included and get transport file for receivers
+        if resource == 'senders':
+            resource_type = 'sender'
+            response_data['receiver_id'] = request_json.get('receiver_id')
+
+            # Update subscription for POST to mock registry
+            subscription_update = resource_data['sender']
+            subscription_update['subscription']['active'] = request_json.get('master_enable',
+                                                                             activations['staged']['master_enable'])
+            subscription_update['version'] = NMOSUtils.get_TAI_time()
+
+            if subscription_update['subscription']['active'] is True:
+                receiver_id = request_json.get('receiver_id', activations['staged']['receiver_id'])
+                subscription_update['subscription']['receiver_id'] = receiver_id
+            else:
+                subscription_update['subscription']['receiver_id'] = None
+
+            default_params['destination_ip'] = activations['transport_params'][0]['destination_ip']
+
+        elif resource == 'receivers':
+            resource_type = 'receiver'
+            response_data['sender_id'] = request_json.get('sender_id')
+            response_data['transport_file'] = request_json.get('transport_file', {'data': None, 'type': None})
+
+            # Update subscription for POST to mock registry
+            subscription_update = resource_data['receiver']
+            subscription_update['subscription']['active'] = request_json.get('master_enable',
+                                                                             activations['staged']['master_enable'])
+            subscription_update['version'] = NMOSUtils.get_TAI_time()
+
+            if subscription_update['subscription']['active'] is True:
+                sender_id = request_json.get('sender_id', activations['staged']['sender_id'])
+                subscription_update['subscription']['sender_id'] = sender_id
+            else:
+                subscription_update['subscription']['sender_id'] = None
+
+        # Get other request data
+        response_data['activation'] = request_json.get('activation', {"activation_time": None, "mode": None,
+                                                                      "requested_time": None})
+        response_data['master_enable'] = request_json.get('master_enable', True)
+
+        if not request_json.get('activation'):
+            # Just staging so return data
+            pass
+
+        elif request_json.get('master_enable') is False:
```

Comment:
Similarly, what does the expression `request_json.get('master_enable') is False` equate to if `master_enable` isn't present in the request?

Reply:
Yes, but your `else` is then the activating case, isn't it?!

Reply:
Yes? It checks if it's just staging, then if it's deactivating, if neither of those conditions are met then it must be activating.

Reply:
I guess so, I'll change it. I didn't realise you could/would stage a deactivation

Reply:
Will look into this more tomorrow, that causes several IS0501 test failures